### PR TITLE
Multithreading

### LINKED ISSUE #6: Multithread movements
There are two different components that need to be developed for this issue:  

- [x] The calls to move the actuators on the x and y device have to be injected in two different threads
- [ ] If either of them fails, the calling function needs to catch the error and log it

Look at the commented out code in `code/controller/actuator.cpp` for reference. To catch the error(s), you can either use exceptions (less safe) or look into how `std::future`, `std::promise`, and/or `std::async` works (the latter being the proper way to do it).

### PR DESCRIPTION
**Pull request by wyxzou**
Closes #6.

### CODE REVIEW

File: code/controller/actuator.cpp [hunk 1]
```diff
@@ -22,152 +23,148 @@ Actuator::Actuator(const QString& serial_port,
...
+    }
+    catch (std::exception& e) {
+        Logger::log(e.what(), Logger::ERROR);
+        success = false;
+    }
+
+    if (success) {
+        Logger::log("Moved { " + std::to_string(dir.x_comp) + ", " + std::to_string(dir.y_comp) + " } in " + std::to_string(t) + " milliseconds.", Logger::INFO);
+        Logger::log("Moved { " + std::to_string(dir.x_comp) + ", " + std::to_string(dir.y_comp) + " } in " + std::to_string(t) + " milliseconds.", Logger::INFO);
+    }
```
Comment (sadmansk) [inline code/controller/actuator.cpp]: Why are you logging the same thing twice? Or am I missing something?

File: code/controller/actuator.cpp [hunk 2]
```diff
@@ -22,152 +23,148 @@ Actuator::Actuator(const QString& serial_port,
...
 }
 
 Actuator::~Actuator() {
     delete m_serial_port;
 }
 
-void Actuator::move(Dir dir, int time) {
-	move(Controller::toVec2(dir), time);
+void Actuator::move(Dir dir, int t) {
+    move(Controller::toVec2(dir), t);
```
Comment (sadmansk) [inline code/controller/actuator.cpp]: Why exactly did you change it? Very poor choice for variable name, remember, for code to be maintainable and in good taste, it should be readable, maintainable and the purpose as obvious as can be at first look.

**Comment by sadmansk** [comment]
Rebase your branch. I'll wait on your addressing of my comments.

File: code/controller/actuator.cpp (hunk 2 above)
Comment (wyxzou) [inline code/controller/actuator.cpp]: time is already a keyword in C++.

Comment (wyxzou) [inline code/controller/actuator.cpp]: I'll change it to something else

Comment (sadmansk) [inline code/controller/actuator.cpp]: `time` isn't a keyword, but I'm assuming you are confusing that with the `time()` function in ctime, but it is global namespaced (we use it like this: `::time(...)`. So having a local variable name `time` is okay.

Comment (wyxzou) [inline code/controller/actuator.cpp]: Hmm... Then I'm not sure why, but we had an error with time when we tried to compile it. Changing it to t seemed to solve it.

Comment (sadmansk) [inline code/controller/actuator.cpp]: that's interesting, which compiler were you using? I have tested on both gcc and msvc and none of them reports any error.

**Comment by sadmansk** [comment]
@wyxzou you still haven't rebased your branch against develop, please do that, otherwise I can't merge.

**Pull request closed without merging**